### DISCUSSION TITLE
Docker containers seem to restart endless

### DISCUSSION THREAD

**Discussion by felixKroni** [telekom-security/tpotce#1583] [Q&A] [answered]
Hi, Im new to this whole thing but I got the newest tpot version installed on debian 12 and the docker container start when i start the server but it seems like they shutdown after some time. I can get to the Menu at https://<ip>:64297 and also open the map even if its empty but i cant access /kibana or /spiderfoot or any other service.

I tried get info about the containers using dps.sh from the docu but i cant find that file. Did it got removes in this version?
I tried to look at the containers using docker ps and docker ps -a

Sometimes there are a lot running:
![image](https://github.com/telekom-security/tpotce/assets/32167792/a30ee069-cfaa-4d8e-8e2c-b98e39faa7fa)

and sometimes only one is running:
![image](https://github.com/telekom-security/tpotce/assets/32167792/51a3cda5-fefd-4128-a27f-86c70a5838d2)

Ive seen in other discussions from last year that ddosspot could be the problem but i dont know how to remove it from the tpot.yml and if that is even a thing anymore.

Thanks for your help

**Comment by felixKroni** [comment]
Some more information i gathered yesterday: 
the tpotinit container restarts every 60sec like defined in the docker-compose. At this point every container gets started but there is no memory overflow as far as i can see it gets to a max usage of 5GB before it all goes down again after about 8sec.
![image](https://github.com/telekom-security/tpotce/assets/32167792/25d639d3-6ccc-4eeb-9885-dbdc9d0de934)

Sometimes i noticed that they were up longer for like 30sec but thats it and all shuts down again.
The commands in htop change though:
![image](https://github.com/telekom-security/tpotce/assets/32167792/3c4f6bc4-80de-48fd-8daa-5e859b8bc89f)


In this small period of time before the containers get deleted i looked at some of the docker logs but there is nothing like an error or something
Suricata:
![image](https://github.com/telekom-security/tpotce/assets/32167792/9d40451c-7677-4068-b537-79c98c2905da)

Cowrie:
![image](https://github.com/telekom-security/tpotce/assets/32167792/6ba842a4-59cc-4064-8cae-058354d9c0e1)

Elasticsearch:
![image](https://github.com/telekom-security/tpotce/assets/32167792/55b1f672-66c7-4478-92b8-e90bec8ba8ee)

**Reply by felixKroni** [reply]
newsflash, it might be a memory problem...
Ive created a custom docker-compose only containing some containers and i get a lot of exit code 137 indicating that the system ran out of memory:
![image](https://github.com/telekom-security/tpotce/assets/32167792/83e41384-a0a4-4a2f-8045-b1596ced0201)
![image](https://github.com/telekom-security/tpotce/assets/32167792/d19638d3-f736-4c97-b551-afa708d210c9)

Although i have no clue why there is a "User-initiated shutdown"

**Comment by barrydekong** [comment]
Hi, I have the same issue even though I reinstalled and update the Tpot again! Anyone can help! thanks.

**Comment by felixKroni** [comment, marked as answer]
I have solved it for myself by using the ubuntu iso that is provided. That worked pretty straightforward without any Problem. I still have no clue what the problem was using Debian.

**Reply by barrydekong** [reply]
Thank you!

**Comment by FloppyDucks** [comment]
I think i know ur issue, glad its working , im pretty sure it is related to the ports that the docker compose file uses when setting up the TPOT, you need to change the default docker compose because a few services use the same ports like 443 and a few others

**Reply by barrydekong** [reply]
Thank you for your information! I have installed Ubuntu with the same issue and have followed your suggestion to edit the docker-compose.yml file but no clue to change the default setting. Grateful if you can guild how to resolve docker compose file, many thanks.

**Comment by RytterMohn** [comment]
hello! I think i know how to solve this problem.
That's because the ports has been used.
On Debian,port 631 are allocated to 'cups-brow server'.
 
I use  
```bash
sudo journalctl -u docker -f
```
to see what happened to the docker.
And I found the port 631 has been allocated，that's why the docker containers restart endless.
I use 
```bash
sudo lsof -i :631
```
found 'cups server' are using 631 port.
so i use
```bash
sudo systemctl stop cups
```
to stop 'cups server'
and then use
```bash
systemctl start tpot
```
to start tpot.
You will see the tpot run normally

![GP_TY)3MMYQ`(APE%MW_6NK](https://github.com/user-attachments/assets/8e7e2cbd-ad9a-4da2-a61e-a3abe5b71300)

(I am not native English speaker, please excuse my poor English)

**Reply by RytterMohn** [reply]
I use debian system and use ./install.sh to install tpot.It works on my machine. if your docker still restart endless,maybe other port have problem(Sometimes port 19 will be allocated to other server)

**Reply by barrydekong** [reply]
It works, thank you for your information!

**Comment by zaterio** [comment]
After a fresh install of Debian 12 and tpot in sensor mode, containers kept restarting, which was due to the WEB_USER variable not being configured in the .env, after adding a dummy value and restarting, the problem was solved

```
docker compose -f  docker-compose.yml up

tpotinit             | 
tpotinit             | # Updating permissions ...
tpotinit             | 
tpotinit             | chmod: /data/nginx/conf: No such file or directory
tpotinit             | chmod: /data/nginx/cert: No such file or directory
tpotinit             | 
tpotinit             | # Checking if OSType is set correctly.
tpotinit             | 
tpotinit             | # Error: WEB_USER is not set or empty. Please check T-Pot .env config.
tpotinit             | 
tpotinit             | # Aborting
```

**Reply by t3chn0m4g3** [reply]
Interesting, this check should only happen if installation is set to Hive.

Is it possible you used the `sensor.yml` but did not adjust `.env` to sensor mode (the deploy script usually takes care if this when run).

https://github.com/telekom-security/tpotce/blob/175e1944c2fe33cb6742d481a494518d565081d4/docker/tpotinit/dist/entrypoint.sh#L223
